packages/react/src/primitives/Pagination/usePagination.ts

```diff
@@ -36,8 +36,8 @@ export const usePagination = (
     }
   }, [currentPage]);
 
-  const onChange = React.useCallback((newPage: number, _: number) => {
-    setCurrentPage(newPage);
+  const onChange = React.useCallback((newPageIndex?: number) => {
```

Comment:
Yea I might need to rethink this. It is weird which parts are potentially undefined in which places..

Reply:
Ok redid the logic a bit. TBH it is still not great because we have a single `PaginationItem` component that renders all parts of the Pagination component and does a bit too much, and the event handlers were really weird. We should do a larger refactor, but at least it makes TS happy now :P